url8af84d296f_mrcosta_gantt_generator_tgz-pJ8-org_gantt_generator_mrcpsp_RunJ8/src/org/swiftgantt/common/Time.java

```diff
@@ -0,0 +1,501 @@
+package org.swiftgantt.common;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * TODO
+ * Extended GregorianCalendar with more common operation
+ * <li>Retrieve partial of time, <code>getYear(), getMonth(), getDay(), getHour(), getMinute(), getSecond()</code>
+ * <li>Truncate time by kinds of time units.
+ * <li>Increase and decrease time by time units.
+ * <li><code>setToFirstDayOfThisWeek(), getFirstDayOfThisWeek(), setToLastDayOfThisWeek(), getLastDayOfThisWeek()</code>
+ * @author Wang Yuxing
+ */
+public class Time extends GregorianCalendar {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class contains 19 clusters of instance methods that do not have any accessed class members in common. For example, a cluster might have 2 methods that access only the class fields x and y, and another cluster might have 2 other methods that access only the class fields a and b. By comparison, 99% of the classes in the CodeGuru reference dataset contain fewer clusters. A high number of these clusters indicates low class cohesion. Classes with low class cohesion contain unrelated operations which make them difficult to understand and less likely to be used. We recommend that you simplify this class or break it into multiple cohesive classes.